malcolm/modules/ADCore/parts/detectordriverpart.py

```diff
@@ -66,10 +68,14 @@ def configure(self,
                 generator.duration)
         self.actions.setup_detector(
             context, completed_steps, steps_to_do, **kwargs)
+        child = context.block_view(self.mri)
+        try:  # todo is this the best way to check if the block has triggerMode ??
+            self.is_hardware_triggered = child.triggerMode.value not in self.soft_trigger_modes
```

Comment:
Do we need the try catch here? The child should be an areaDetector, so will always have triggerMode...

Reply:
xmapdriverpart and reframeplugin both derive from detectordriverpart and do not have triggerMode. This try block means that these will always be 'hardware triggered' meaning that configure() will start them. 

I think that is correct behaviour for reframeplugin but not sure about xmap (not sure why it has no trigger mode?)

Reply:
how about we only do this check if we have some soft trigger modes? Otherwise we default to hardware triggered. Then we would never do the check for xmap or the reframe plugin

Reply:
Currently soft_trigger_modes defaults to ('Internal',). We would have to require that subclasses of detectordriverpart pass soft_trigger_modes=None if they do not have triggerMode (which is unintuitive). So we can change the default to None and require all detector types pass a soft_trigger_modes or continue to use the more direct approach of testing if triggerMode exists.

Reply:
agreed on soft_trigger_modes defaults to None. committed changes.